benchmark/usd/usd_cost_estimate.ts

```diff
@@ -0,0 +1,188 @@
+import { CYCLE_COST_TABLE, USD_PER_CYCLE } from './cycle_costs';
+import { UsageConfig } from './usage_config';
+
+const SECONDS_PER_YEAR = 365 * 24 * 60 * 60;
+
+export type USDCostEstimates = {
+    azle: USDCostEstimateUsageGroup;
+    motoko: USDCostEstimateUsageGroup;
+    rust: USDCostEstimateUsageGroup;
+};
+
+type USDCostEstimateUsageGroup = {
+    light: USDCostEstimateHeavinessGroup;
+    moderate: USDCostEstimateHeavinessGroup;
+    heavy: USDCostEstimateHeavinessGroup;
+};
+
+type USDCostEstimateHeavinessGroup = {
+    query_update_even: USDCostEstimate;
+    query_heavy: USDCostEstimate;
+    update_heavy: USDCostEstimate;
+};
+
+type USD = number;
+type Cycles = number;
+
+export type USDCostEstimate = {
+    ingress_messages: USD;
+    ingress_bytes_query_messages: USD;
+    ingress_bytes_update_messages: USD;
+    update_messages: USD;
+    update_instructions: USD;
+    xnet_calls: USD;
+    xnet_call_bytes: USD;
+    gb_storage: USD;
+    total_cost: USD;
+    // TODO Do we want these here? It might be good enough to just let the users figure this out on their own
+    // azle_motoko_change_multiplier: number;
+    // azle_rust_change_multiplier: number;
+    // motoko_azle_change_multiplier: number;
+    // motoko_rust_change_multiplier: number;
+    // rust_azle_change_multiplier: number;
+    // rust_motoko_change_multiplier: number;
+};
+
+export function get_usd_cost_estimate(
+    usage_config: UsageConfig,
+    wasm_instructions_per_update_message: number
+): USDCostEstimate {
+    const ingress_messages_cost_usd: USD =
+        calculate_cost_usd_ingress_messages(usage_config);
+    const ingress_bytes_query_messages_cost_usd: USD =
+        calculate_cost_usd_ingress_bytes_query_messages(usage_config);
+    const ingress_bytes_update_messages_cost_usd: USD =
+        calculate_cost_usd_ingress_bytes_update_messages(usage_config);
+    const update_messages_cost_usd: USD =
+        calculate_cost_usd_update_messages(usage_config);
+    const update_instructions_cost_usd: USD =
+        calculate_cost_usd_update_instructions(
+            usage_config,
+            wasm_instructions_per_update_message
+        );
+    const xnet_calls_cost_usd: USD =
+        calculate_cost_usd_xnet_calls(usage_config);
+    const xnet_bytes_cost_usd: USD =
+        calculate_cost_usd_xnet_bytes(usage_config);
+    const gb_storage_cost_usd: USD =
+        calculate_cost_usd_gb_storage(usage_config);
+    const total_cost_usd: USD =
+        ingress_messages_cost_usd +
+        ingress_bytes_query_messages_cost_usd +
+        ingress_bytes_update_messages_cost_usd +
+        update_messages_cost_usd +
+        update_instructions_cost_usd +
+        gb_storage_cost_usd;
```

Comment:
@lastmjs should `xnet_calls_cost_usd` and `xnet_bytes_cost_usd` be included in `total_cost_usd`?